lib/Dialect/Seq/Transforms/ExternalizeClockGate.cpp

```diff
@@ -61,43 +62,54 @@ void ExternalizeClockGatePass::runOnOperation() {
     modulePorts.push_back({{builder.getStringAttr(testEnableName), i1Type,
                             ModulePort::Direction::Input}});
 
-  auto moduleOp = builder.create<HWModuleExternOp>(
+  auto externModuleOp = builder.create<HWModuleExternOp>(
       getOperation().getLoc(), builder.getStringAttr(moduleName), modulePorts,
       moduleName);
-  symtbl.insert(moduleOp);
+  symtbl.insert(externModuleOp);
 
   // Replace all clock gates with an instance of the external module.
   SmallVector<Value, 4> instPorts;
-  for (auto ckgOp : clockGatesToReplace) {
-    ImplicitLocOpBuilder builder(ckgOp.getLoc(), ckgOp);
-
-    Value enable = ckgOp.getEnable();
-    Value testEnable = ckgOp.getTestEnable();
-
-    // If the clock gate has a test enable operand but the module does not, add
-    // a `comb.or` to merge the two enable conditions.
-    if (hasTestEnable && !testEnable)
-      testEnable = builder.create<ConstantOp>(i1Type, 0);
-
-    // If the clock gate has no test enable operand but the module does, add a
-    // constant 0 input.
-    if (!hasTestEnable && testEnable) {
-      enable = builder.createOrFold<comb::OrOp>(enable, testEnable, true);
-      testEnable = {};
+  for (auto &[module, clockGatesToReplace] : gatesInModule) {
+    Value cstFalse;
+    for (auto ckgOp : clockGatesToReplace) {
+      ImplicitLocOpBuilder builder(ckgOp.getLoc(), ckgOp);
+
+      Value enable = ckgOp.getEnable();
+      Value testEnable = ckgOp.getTestEnable();
+
+      // If the clock gate has a test enable operand but the module does not,
+      // add a `comb.or` to merge the two enable conditions.
+      if (hasTestEnable && !testEnable) {
+        if (!cstFalse) {
+          cstFalse = builder.create<ConstantOp>(i1Type, 0);
+        }
+        testEnable = cstFalse;
+      }
+
+      // If the clock gate has no test enable operand but the module does, add a
+      // constant 0 input.
+      if (!hasTestEnable && testEnable) {
+        enable = builder.createOrFold<comb::OrOp>(enable, testEnable, true);
+        testEnable = {};
+      }
+
+      instPorts.push_back(ckgOp.getInput());
+      instPorts.push_back(enable);
+      if (testEnable)
+        instPorts.push_back(testEnable);
+
+      auto instOp = builder.create<InstanceOp>(
+          externModuleOp, builder.getStringAttr(instName), instPorts);
+
+      if (auto sym = ckgOp.getInnerSymAttr())
```

Comment:
If you'd like, could collapse this into the `InstanceOp` creation, although would need to provide empty `parameters`.